cocos/platform/CCFileUtils.h

```diff
@@ -41,6 +42,66 @@ NS_CC_BEGIN
  * @{
  */
 
+
+class ResizableBuffer {
+public:
+    virtual ~ResizableBuffer() {}
+    virtual void resize(size_t size) = 0;
+    virtual void* buffer() const = 0;
+};
+
+template<typename T>
+class ResizableBufferAdapter { };
+
+
+template<typename C, typename T, typename A>
+class ResizableBufferAdapter< std::basic_string<C, T, A> > : public ResizableBuffer {
+    typedef std::basic_string<C, T, A> BufferType;
+    BufferType* _buffer;
+public:
+    explicit ResizableBufferAdapter(BufferType* buffer) : _buffer(buffer) {}
+    virtual void resize(size_t size) override {
+        _buffer->resize((size + sizeof(C) - 1) / sizeof(C));
```

Comment:
`ResizableBuffer::resize()` is not for user calling, `FileUtils::getContents()` calls it internally:

1. `ResizableBuffer::resize()` is called by `` at cocos/platform/CCFileUtils.cpp` [line 687](https://github.com/cocos2d/cocos2d-x/pull/15479/files#diff-cb65223cd3edf30671fefad4a1883edfR687) in `FileUtils::getContents()` when the file size is determined, and thus, the file size is passed to `ResizableBuffer::resize()`.
2. `ResizableBufferAdapter` convert bytes to number of elements need.

Example Usage:

```C++
    std::string<char32_t> buf;
    fs->getContents(file, &buf); // resize buf automatically inside getContents()
    std::cout << "I got " << buf.size() << " of wchar32_t" << std::endl;
```